Sources/Widgets/Widgets3D/LineWidget/behavior.js

```diff
@@ -109,6 +176,69 @@ export default function widgetBehavior(publicAPI, model) {
     }
   };
 
+  publicAPI.toggleHandleVisibility = (bv) => {
+    if (bv === HandleVisibilityBehavior.TOGGLE_ON) {
+      if (handle1Visibility === false && handle1.getActive()) {
+        model.representations[0].setHandleVisibility(true);
+        model.handle1ToHide = true;
+      }
+      if (handle2Visibility === false && handle2.getActive()) {
+        model.representations[1].setHandleVisibility(true);
+        model.handle2ToHide = true;
+      }
+      return;
+    }
+    if (model.handle1ToHide === true) {
+      model.handle1ToHide = false;
+      model.representations[0].setHandleVisibility(handle1Visibility);
+    }
+    if (model.handle2ToHide === true) {
+      model.handle2ToHide = false;
+      model.representations[1].setHandleVisibility(handle2Visibility);
+    }
+  };
+
+  publicAPI.isHandleGlyph2D = (handleShape) => {
+    return (
+      handleShape === HandleRepresentationType.ARROWHEAD3 ||
+      handleShape === HandleRepresentationType.ARROWHEAD4 ||
+      handleShape === HandleRepresentationType.ARROWHEAD6 ||
+      handleShape === HandleRepresentationType.STAR
+      // TODO: implement rotation for circleHandle
+    );
+  };
+
+  publicAPI.setRotationHandleToFaceCamera = () => {
```

Comment:
instead of having to call it manually, maybe you should observe when the shape is modified.